src/hotspot/share/logging/logConfiguration.cpp

```diff
@@ -253,6 +268,15 @@ void LogConfiguration::configure_output(size_t idx, const LogSelectionList& sele
     on_level[level]++;
   }
 
+  // for async logging, enqueuing instead of writing is still under protection of the synchronization
+  // `wait_until_no_readers()`. There are 2 hazards in async logging as follows.
+  //  1. asynclog buffer may be holding some log messages with previous decorators.
+  //  2. asynclog buffer may be holding some log messages targeting to the output 'idx'.
+  //
+  // A flush operation guarantees to all pending messages in buffer are written before returning. Therefore,
+  // the two hazards won't appear after it. It's a nop if async logging is not set.
```

Comment:
Given all the other comments I think this can be simplified:

// For async logging we have to ensure that all enqueued messages, which may refer to previous decorators,
// or a soon-to-be-deleted outputs, are written out first. The flush() call ensures this. It is a no-op
// if async logging is not enabled.